Source/CSharpEssentials/GetterOnlyAutoProperty/UseGetterOnlyAutoPropertyAnalyzer.cs

```diff
@@ -1,35 +1,214 @@
+            }
+        }
+
+        private static bool IsIdentifierWithinAnAssignmentButNotInAConstructor(SyntaxNode identifier, INamedTypeSymbol containingType, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            // Is the property being updated but not within the constructor?
+            var node = identifier;
+            if (!IsUpdatingExpression(ref node))
+            {
+                return false;
+            }
+
+            return !IsWithinConstructorOf(node.Parent, containingType, semanticModel, cancellationToken);
+        }
+
+        private static HashSet<ISymbol> GetAutoPropsWithPrivateSetters(INamedTypeSymbol type, CancellationToken cancellationToken)
+        {
+            HashSet<ISymbol> candidates = null;
+
+            var allProperties = type.GetMembers().Where(s => s.Kind == SymbolKind.Property);
+            foreach (var property in allProperties)
+            {
+                var propertySymbol = (IPropertySymbol)property;
+                if (!propertySymbol.IsReadOnly && propertySymbol.ExplicitInterfaceImplementations.IsDefaultOrEmpty)
+                {
+                    var setMethod = propertySymbol.SetMethod;
+                    if (setMethod != null && setMethod.DeclaredAccessibility == Accessibility.Private)
+                    {
+                        // Find the syntax for the setter.
+                        var declaration = setMethod.DeclaringSyntaxReferences.FirstOrDefault().GetSyntax(cancellationToken) as AccessorDeclarationSyntax;
+                        if (declaration != null && declaration.Body == null)
+                        {
+                            // An empty body indicates it's an auto-prop
+                            (candidates ?? (candidates = new HashSet<ISymbol>())).Add(propertySymbol);
+                        }
+                    }
+                }
+            }
+
+            return candidates;
+        }
+
+        private static bool IsDescendant(SyntaxNode root, SyntaxNode node)
+        {
+            while (node != null)
+            {
+                if (node == root)
+                {
+                    return true;
+                }
+
+                node = node.Parent;
+            }
 
-            if (accessorDeclaration.Body == null &&
-                !accessorDeclaration.SemicolonToken.IsMissing &&
-                accessorDeclaration.Modifiers.Any(SyntaxKind.PrivateKeyword))
+            return false;
+        }
+
+        private static bool IsUpdatingExpression(ref SyntaxNode node)
+        {
+            var identifier = node;
+            for (node = node.Parent; node != null; node = node.Parent)
             {
-                context.ReportDiagnostic(
-                    Diagnostic.Create(
-                        DiagnosticDescriptors.UseGetterOnlyAutoProperty,
-                        accessorDeclaration.GetLocation()));
+                switch (node.Kind())
```

Comment:
For the simple and compound assignment, you should do the same thing, which is to bind the Left value of the assignment and verify that it is the same property symbol.